### PR TITLE
Add mypy Pre-Commit For hi & hit Pt.10

### PR DESCRIPTION
**Pull request by daralynnrhode**
# Adding the mypy Pre-commit hook for hi & hit.
## Overview
Changes for this PR will occur in the `imap_processing/hi` and `imap_processing/hit` files . Applying the mypy pre-commit and making any necessary changes to then pass the checks. 

Please double check all function definitions and the types applied to the parameters. 
With any `TODO` listed I tried to explain what error it was causing and then ignored the error in the code line. 

The places where inline ignore comments are will be changed/fixed  before the PR is merged unless for special instances. 

This one is kinda long because I could not exclude one folder without excluding the other from the checks. Sorry in advance.

## Updated Files
-imap_processing/hit/l1a/hit_l1a.py
-imap_processing/hi/utils.py
-imap_processing/hit/l0/utils/hit_base.py
-imap_processing/hi/l1a/science_direct_event.py
-imap_processing/hi/l0/decom_hi.py
-imap_processing/hit/l0/data_classes/housekeeping.py
-imap_processing/hi/l1b/hi_l1b.py
-imap_processing/hi/l1a/hi_l1a.py
-imap_processing/hit/l1b/hit_l1b.py
-imap_processing/hit/l1a/hit_l1a.py

### CODE REVIEW

File: imap_processing/hi/l1a/science_direct_event.py [hunk 1]
```diff
@@ -107,14 +107,18 @@ def parse_direct_event(event_data: str) -> dict:
     metaevent = 0
     if event_type == metaevent:
         # parse metaevent
-        event_type = event_data[:2]
+        event_type = event_data[:2]  # type: ignore[assignment]
+        # TODO Change, Incompatible types in assignment
+        # (expression has type "str", variable has type "int")
```
Comment (tmplummer) [inline imap_processing/hi/l1a/science_direct_event.py]: I suggest changing the initial declaration of `event_type` to `event_type = int(event_data[:2], 2)`. Then this line reassigning `event_type` can be removed.

```suggestion
```

File: imap_processing/hi/l1a/science_direct_event.py [hunk 2]
```diff
@@ -107,14 +107,18 @@ def parse_direct_event(event_data: str) -> dict:
...
         esa_step = event_data[2:6]
         subseconds = event_data[6:16]
         seconds = event_data[16:]
 
         # return parsed metaevent data
         return {
-            "start_bitmask_data": int(event_type, 2),
+            "start_bitmask_data": int(event_type, 2),  # type: ignore[call-overload]
+            # TODO Change, No overload variant of "int"
+            # matches argument types "int", "int"
```
Comment (tmplummer) [inline imap_processing/hi/l1a/science_direct_event.py]:
```suggestion
            "start_bitmask_data": event_type
```

**Review by tmplummer — APPROVED** [review]
Thanks for doing this. I am learning how sloppy my type specifications are from this PR. This is valuable work for the project!

With the recommended changes to fix TODOs, the changes for Hi look good to me.

**Review by sdhoyt — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**